cray_product_catalog/catalog_update.py

```diff
@@ -283,7 +304,20 @@ def main():
     if VALIDATE_SCHEMA:
         validate_schema(data)
 
-    update_config_map(data, CONFIG_MAP, CONFIG_MAP_NAMESPACE)
+    product_config_map = format_product_cm_name(CONFIG_MAP, PRODUCT)
+
+    LOGGER.debug("Splitting cray-product-catalog data")
+    main_cm_data, prod_cm_data = split_catalog_data(data)
+
+    if prod_cm_data and product_config_map == '':
+        LOGGER.error("Not updating ConfigMaps because the provided product name is invalid: '%s'", PRODUCT)
+        raise SystemExit(1)
+
+    update_config_map(main_cm_data, CONFIG_MAP, CONFIG_MAP_NAMESPACE)
```

Comment:
If the loop in `update_config_map` exits due to exceeding `retries`, this will not log an error. Update the logic to log an error and exit with non-zero status.

Reply:
Added log message that max retires attempted if update_config_map exits due to exceeding retries